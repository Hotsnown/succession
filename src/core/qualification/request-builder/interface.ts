
export interface Attributes {
}

export interface Family {
    member_id: string;
    childs: string[];
    attributes: Attributes;
}

export interface Query {
    de_cujus: string;
    family: Family[];
}
