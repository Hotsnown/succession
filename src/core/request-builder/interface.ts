
export interface Data {
}

export interface Family {
    member_id: string;
    childs: string[];
    data: Data;
}

export interface Query {
    de_cujus: string;
    family: Family[];
}
