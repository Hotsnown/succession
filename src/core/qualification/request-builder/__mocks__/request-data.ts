import { Query } from '../interface'

export const requestattributes: Query = {"de_cujus":"Pierre","family":[
    {
        "member_id":"Pierre",
        "childs":[],
        "attributes":{"status": 'valid'}
    },
    {"member_id":"Marie","childs":[],"attributes":{"status": 'valid'}},
    {"member_id":"Fred","childs":["Pierre","Marie"],"attributes":{"status": 'valid'}},
    {"member_id":"Bea","childs":["Armand","Morgane","Thibault","Romain"],"attributes":{"status": 'valid'}},
    {"member_id":"Armand","childs":[],"attributes":{"status": 'valid'}},
    {"member_id":"Morgane","childs":[],"attributes":{"status": 'valid'}},
    {"member_id":"Thibault","childs":[],"attributes":{"status": 'valid'}},
    {"member_id":"Romain","childs":[],"attributes":{"status": 'valid'}},
    {"member_id":"Sylvain","childs":[],"attributes":{"status": 'valid'}},
    {"member_id":"Steph","childs":[],"attributes":{"status": 'valid'}},
    {"member_id":"Bernard","childs":["Fred","Sylvain","Bea","Steph"],"attributes":{"status": 'valid'}},
    {"member_id":"Claudine","childs":["Fred","Sylvain","Bea","Steph"],"attributes":{"status": 'valid'}}]}