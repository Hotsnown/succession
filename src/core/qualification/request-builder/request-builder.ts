import { Query } from './interface';
const fetch = require('node-fetch')

async function getQualificationFrom (query: Query) {

    const requestOptions = {
      method: 'POST',
      headers: {"Content-Type": "application/json"},
      body: JSON.stringify(query)
    };
    
    const result  = await fetch("http://127.0.0.1:5000/todo/api/v1.0/tasks", requestOptions)
      .then((response: Response) => response.json())
      .then((result: string) => {return result})
      .catch(() => alert('Server has not been found. Please set it up with "npm run python".'))
    ;

    return result
}

export default getQualificationFrom