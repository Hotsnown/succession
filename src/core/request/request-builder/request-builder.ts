import { Query } from './interface';
const fetch = require('node-fetch')

async function getQualificationFrom (query: Query) {

    const requestOptions = {
      method: 'POST',
      headers: {"Content-Type": "application/json"},
      body: JSON.stringify(query)
    };
    
    const result  = await fetch("http://127.0.0.1:5000/todo/api/v1.0/tasks", requestOptions)
      .then((response: Response) => response.json())
      .then((result: string) => {return result})
      .catch((error: Error) => console.log('error', error));

    return result
}

export default getQualificationFrom