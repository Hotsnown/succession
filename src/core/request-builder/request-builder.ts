import { requestData } from './request-data'
const fetch = require('node-fetch')

export async function requestBuilder () {

    const raw = JSON.stringify(requestData);
    
    const requestOptions = {
      method: 'POST',
      headers: {"Content-Type": "application/json"},
      body: raw,
    };
    
    const result  = await fetch("http://127.0.0.1:5000/todo/api/v1.0/tasks", requestOptions)
      .then((response: Response) => response.json())
      .then((result: string) => {return result})
      .catch((error: Error) => console.log('error', error));

    return result
}
