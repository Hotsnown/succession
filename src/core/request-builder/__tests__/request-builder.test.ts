/**
 * @jest-environment node
 */
import { getQualificationFrom } from '../request-builder'
import { requestData } from './request-data'

it('should return a match the snapshot', async () => {
    
    expect(true).toBeTruthy()
    //const data = await getQualificationFrom(requestData)
    //expect(data).toMatchSnapshot()
})