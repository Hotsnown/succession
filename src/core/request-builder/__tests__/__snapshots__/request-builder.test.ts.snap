// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return a match the snapshot 1`] = `
Object {
  "task": Array [
    Object {
      "childs": Array [],
      "data": Object {
        "degre": 0,
        "ordre": 0,
      },
      "member_id": "Pierre",
    },
    Object {
      "childs": Array [],
      "data": Object {
        "degre": 2,
        "ordre": 2,
      },
      "member_id": "Marie",
    },
    Object {
      "childs": Array [
        "Pierre",
        "Marie",
      ],
      "data": Object {
        "degre": 1,
        "ordre": 3,
      },
      "member_id": "Fred",
    },
    Object {
      "childs": Array [
        "Armand",
        "Morgane",
        "Thibault",
        "Romain",
      ],
      "data": Object {
        "degre": 2,
        "ordre": 4,
      },
      "member_id": "Bea",
    },
    Object {
      "childs": Array [],
      "data": Object {
        "degre": 4,
        "ordre": 4,
      },
      "member_id": "Armand",
    },
    Object {
      "childs": Array [],
      "data": Object {
        "degre": 4,
        "ordre": 4,
      },
      "member_id": "Morgane",
    },
    Object {
      "childs": Array [],
      "data": Object {
        "degre": 4,
        "ordre": 4,
      },
      "member_id": "Thibault",
    },
    Object {
      "childs": Array [],
      "data": Object {
        "degre": 4,
        "ordre": 4,
      },
      "member_id": "Romain",
    },
    Object {
      "childs": Array [],
      "data": Object {
        "degre": 2,
        "ordre": 4,
      },
      "member_id": "Sylvain",
    },
    Object {
      "childs": Array [],
      "data": Object {
        "degre": 2,
        "ordre": 4,
      },
      "member_id": "Steph",
    },
    Object {
      "childs": Array [
        "Fred",
        "Sylvain",
        "Bea",
        "Steph",
      ],
      "data": Object {
        "degre": 2,
        "ordre": 3,
      },
      "member_id": "Bernard",
    },
    Object {
      "childs": Array [
        "Fred",
        "Sylvain",
        "Bea",
        "Steph",
      ],
      "data": Object {
        "degre": 2,
        "ordre": 3,
      },
      "member_id": "Claudine",
    },
  ],
}
`;
